INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"Dolev"}
INFO:root:New user connected: Dolev
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f87e8913010>, 1705072494.553424)}
INFO:root:Received message from the client: {"type":"message","userName":"Dolev","text":"heyyyyyy"}
INFO:root:Closed connection with None
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"Dolev"}
INFO:root:New user connected: Dolev
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fc1d0913d30>, 1705072551.0046442)}
INFO:root:Received message from the client: {"type":"message","userName":"Dolev","text":"heyyy"}
INFO:websockets.server:connection rejected (400 Bad Request)
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"Shir"}
INFO:root:New user connected: Shir
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fc1d0913d30>, 1705072551.0046442), 'Shir': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fc1d0a10550>, 1705072609.09658)}
INFO:root:Received message from the client: {"type":"message","userName":"Shir","text":"heyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy"}
INFO:root:Received message from the client: {"type":"message","userName":"Shir","text":"how are me "}
INFO:root:Received message from the client: {"type":"message","userName":"Dolev","text":"good"}
INFO:root:Received message from the client: {"type":"message","userName":"Shir","text":"ownload 131100 Animal Avatar Vector Icons for commercial and personal use. Available for free or premium in line, flat, gradient, isometric, glyph, ..."}
INFO:root:Received message from the client: {"type":"message","userName":"Shir","text":"hrh hehhshhs hhhh es se vnjnjnjnjnse jnnnjn\njdfbjv bdjfvjndfjnvj njnjnjnjn jndfjnsjv jndfj nvjdfnjvndfjnvjdfnjvndsjfnvjdfnjvdfjvnsdfv"}
INFO:root:Received message from the client: {"type":"disconnect","username":"Shir"}
INFO:root:Closed connection with Shir
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 61, in handle_client
    await send_to_all_user_disconnect(message_json["username"])
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 37, in send_to_all_user_disconnect
    logging,info(f"User disconnected: {username}")
NameError: name 'info' is not defined
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"disconnect","username":"Dolev"}
INFO:root:Closed connection with Dolev
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 61, in handle_client
    await send_to_all_user_disconnect(message_json["username"])
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 37, in send_to_all_user_disconnect
    logging,info(f"User disconnected: {username}")
NameError: name 'info' is not defined
INFO:websockets.server:connection closed
INFO:websockets.server:connection rejected (400 Bad Request)
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fc1d0a104f0>, 1705072725.952399)}
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"s"}
INFO:root:New user connected: s
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fc1d0a104f0>, 1705072725.952399), 's': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fc1d0a10a30>, 1705072728.8524108)}
INFO:root:Received message from the client: {"type":"disconnect","username":"d"}
INFO:root:Closed connection with d
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 61, in handle_client
    await send_to_all_user_disconnect(message_json["username"])
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 37, in send_to_all_user_disconnect
    logging,info(f"User disconnected: {username}")
NameError: name 'info' is not defined
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"disconnect","username":"s"}
INFO:root:Closed connection with s
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 61, in handle_client
    await send_to_all_user_disconnect(message_json["username"])
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 37, in send_to_all_user_disconnect
    logging,info(f"User disconnected: {username}")
NameError: name 'info' is not defined
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fee9849beb0>, 1705072868.128222)}
INFO:root:Received message from the client: {"type":"message","userName":"d","text":"dfvdfv"}
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"s"}
INFO:root:New user connected: s
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fee9849beb0>, 1705072868.128222), 's': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fee9849abc0>, 1705072883.7572548)}
INFO:root:Received message from the client: {"type":"disconnect","username":"s"}
INFO:root:Closed connection with s
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 61, in handle_client
    await send_to_all_user_disconnect(message_json["username"])
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 37, in send_to_all_user_disconnect
    logging,info(f"User disconnected: {username}")
NameError: name 'info' is not defined
INFO:websockets.server:connection closed
INFO:root:Closed connection with None
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda00563d00>, 1705072920.761652)}
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"s"}
INFO:root:New user connected: s
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda00563d00>, 1705072920.761652), 's': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda005635b0>, 1705072922.9400299)}
INFO:root:Received message from the client: {"type":"disconnect","username":"d"}
INFO:root:User disconnected: d
INFO:root:clients_by_username: {'s': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda005635b0>, 1705072922.9400299)}
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"message","userName":"s","text":"dfvdfv"}
INFO:root:Received message from the client: {"type":"disconnect","username":"s"}
INFO:root:User disconnected: s
INFO:root:clients_by_username: {}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d02e0>, 1705073610.0715232)}
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"s"}
INFO:root:New user connected: s
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d02e0>, 1705073610.0715232), 's': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d0d90>, 1705073663.917182)}
INFO:root:Received message from the client: {"type":"disconnect","username":"s"}
INFO:root:User disconnected: s
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d02e0>, 1705073610.0715232)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"s"}
INFO:root:New user connected: s
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d02e0>, 1705073610.0715232), 's': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d12a0>, 1705073692.799763)}
INFO:root:Received message from the client: {"type":"disconnect","username":"s"}
INFO:root:User disconnected: s
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d02e0>, 1705073610.0715232)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dd"}
INFO:root:New user connected: dd
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d02e0>, 1705073610.0715232), 'dd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1540>, 1705073709.8837051)}
INFO:root:Received message from the client: {"type":"disconnect","username":"d"}
INFO:root:User disconnected: d
INFO:root:clients_by_username: {'dd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1540>, 1705073709.8837051)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:clients_by_username: {'dd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1540>, 1705073709.8837051), 'd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d0fd0>, 1705073723.1709352)}
INFO:root:Received message from the client: {"type":"disconnect","username":"d"}
INFO:root:User disconnected: d
INFO:root:clients_by_username: {'dd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1540>, 1705073709.8837051)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"ffff"}
INFO:root:New user connected: ffff
INFO:root:clients_by_username: {'dd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1540>, 1705073709.8837051), 'ffff': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d0580>, 1705073807.4799418)}
INFO:root:Received message from the client: {"type":"disconnect","username":"ffff"}
INFO:root:User disconnected: ffff
INFO:root:clients_by_username: {'dd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1540>, 1705073709.8837051)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"vdfvdfv"}
INFO:root:New user connected: vdfvdfv
INFO:root:clients_by_username: {'dd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1540>, 1705073709.8837051), 'vdfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1960>, 1705073859.4961028)}
INFO:root:Received message from the client: {"type":"disconnect","username":"dd"}
INFO:root:User disconnected: dd
INFO:root:clients_by_username: {'vdfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1960>, 1705073859.4961028)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"sdsdc"}
INFO:root:New user connected: sdsdc
INFO:root:clients_by_username: {'vdfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1960>, 1705073859.4961028), 'sdsdc': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1e70>, 1705073883.245929)}
INFO:root:Received message from the client: {"type":"disconnect","username":"vdfvdfv"}
INFO:root:User disconnected: vdfvdfv
INFO:root:clients_by_username: {'sdsdc': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1e70>, 1705073883.245929)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"sdsdc"}
INFO:root:New user connected: sdsdc
INFO:root:clients_by_username: {'sdsdc': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda00563520>, 1705074048.0426629)}
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"vvv"}
INFO:root:New user connected: vvv
INFO:root:clients_by_username: {'sdsdc': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda00563520>, 1705074048.0426629), 'vvv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d1b40>, 1705074056.457419)}
INFO:root:Received message from the client: {"type":"disconnect","username":"vvv"}
INFO:root:User disconnected: vvv
INFO:root:clients_by_username: {'sdsdc': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda00563520>, 1705074048.0426629)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dfvdfv"}
INFO:root:New user connected: dfvdfv
INFO:root:clients_by_username: {'sdsdc': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda00563520>, 1705074048.0426629), 'dfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d0700>, 1705074161.9456809)}
INFO:root:Received message from the client: {"type":"disconnect","username":"sdsdc"}
INFO:root:User disconnected: sdsdc
INFO:root:clients_by_username: {'dfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda101d0700>, 1705074161.9456809)}
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"disconnect","username":"dfvdfv"}
INFO:root:User disconnected: dfvdfv
INFO:root:clients_by_username: {}
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"Dolev"}
INFO:root:New user connected: Dolev
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736)}
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"Shir"}
INFO:root:New user connected: Shir
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736), 'Shir': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd90>, 1705074392.08779)}
INFO:root:Received message from the client: {"type":"message","userName":"Dolev","text":"hey"}
INFO:root:Received message from the client: {"type":"message","userName":"Shir","text":"how are you"}
INFO:root:Received message from the client: {"type":"disconnect","username":"Shir"}
INFO:root:User disconnected: Shir
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"G"}
INFO:root:New user connected: G
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736), 'G': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf88c8610>, 1705074430.886132)}
INFO:root:Received message from the client: {"type":"disconnect","username":"G"}
INFO:root:User disconnected: G
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"t"}
INFO:root:New user connected: t
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736), 't': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf88c8ca0>, 1705074447.5528388)}
INFO:root:Received message from the client: {"type":"disconnect","username":"t"}
INFO:root:User disconnected: t
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"t"}
INFO:root:New user connected: t
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736), 't': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf88c91b0>, 1705074451.873409)}
INFO:root:Received message from the client: {"type":"disconnect","username":"t"}
INFO:root:User disconnected: t
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"y"}
INFO:root:New user connected: y
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736), 'y': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf88c9030>, 1705074455.0482519)}
INFO:root:Received message from the client: {"type":"disconnect","username":"y"}
INFO:root:User disconnected: y
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736)}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"shir"}
INFO:root:New user connected: shir
INFO:root:clients_by_username: {'Dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf87cbd30>, 1705074385.818736), 'shir': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffbf88c8a00>, 1705074464.531731)}
INFO:root:Closed connection with shir
INFO:websockets.server:connection closed
INFO:root:Closed connection with None
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:clients_by_username: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ff778913670>, 1705092155.423469)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:clients_by_username: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ff778912b90>, 1705092159.849219)}
INFO:root:Received message from the client: {"type":"message","userName":"dolev","text":"hey"}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:clients_by_username: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ff778a107c0>, 1705092167.282133)}
INFO:root:Received message from the client: {"type":"message","userName":"dolev","text":"dfvdfv"}
INFO:root:Received message from the client: {"type":"message","userName":"dolev","text":"dfvdfvdfv"}
INFO:root:Received message from the client: {"type":"message","userName":"dolev","text":"fvfvvv"}
INFO:root:Received message from the client: {"type":"message","userName":"dolev","text":"dfvdfv"}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"disconnect","username":"dolev"}
INFO:root:User disconnected: dolev
INFO:root:clients_by_username: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:clients_by_username: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ff4b09136a0>, 1705092223.7443688)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"shir"}
INFO:root:New user connected: shir
INFO:root:clients_by_username: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ff4b09136a0>, 1705092223.7443688), 'shir': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ff4b09135e0>, 1705092227.656339)}
INFO:root:Received message from the client: {"type":"message","userName":"shir","text":"erfve"}
INFO:root:Received message from the client: {"type":"message","userName":"dolev","text":"dfvfvv"}
INFO:root:Closed connection with None
INFO:websockets.server:connection closed
INFO:root:Closed connection with None
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"v"}
INFO:root:New user connected: v
INFO:root:clients_by_username: {'v': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd980913010>, 1705092712.593708)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"vv"}
INFO:root:New user connected: vv
INFO:root:clients_by_username: {'v': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd980913010>, 1705092712.593708), 'vv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd980913dc0>, 1705092717.5089111)}
INFO:root:Received message from the client: {"type":"message","userName":"vv","text":"vdfvdfv"}
INFO:root:Received message from the client: {"type":"message","userName":"v","text":"dfvdfv"}
INFO:root:Closed connection with None
INFO:websockets.server:connection closed
INFO:root:Closed connection with None
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd318913010>, 1705093347.9551091)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd318913010>, 1705093347.9551091), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd318913dc0>, 1705093354.089355)}
INFO:root:Received message from the client: {"type":"message","userName":"dv","text":"dfvdfv"}
INFO:root:Received message from the client: {"type":"message","userName":"d","text":"dfvdfvdfv"}
INFO:root:Closed connection with None
INFO:websockets.server:connection closed
INFO:root:Closed connection with None
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 61, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"connect","username":"ddddd"}
INFO:root:New user connected: ddddd
INFO:root:clients_by_username: {'ddddd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ff2580300d0>, 1705137290.562268)}
INFO:root:Received message from the client: {"type":"message","userName":"ddddd","text":"dfvdfvdf"}
INFO:root:Received message from the client: {"type":"message","userName":"ddddd","text":"fvvfvfv"}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 61, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"ssssss"}
INFO:root:New user connected: ssssss
INFO:root:clients_by_username: {'ddddd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ff2580300d0>, 1705137290.562268), 'ssssss': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ff258030b20>, 1705137319.7498398)}
INFO:root:Received message from the client: {"type":"message","userName":"ddddd","text":"dfvdfvdfv"}
INFO:root:Received message from the client: {"type":"message","userName":"ssssss","text":"dfvdfvdfv"}
INFO:root:Received message from the client: {"type":"disconnect","username":"ssssss"}
INFO:root:User disconnected: ssssss
INFO:root:clients_by_username: {'ddddd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ff2580300d0>, 1705137290.562268)}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"disconnect","username":"ddddd"}
INFO:root:User disconnected: ddddd
INFO:root:clients_by_username: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd180917130>, 1705138774.141022)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd180917130>, 1705138774.141022), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd1809f0310>, 1705138774.153314)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 64, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 56, in handle_client
    CLIENTS_BY_USERNAME[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 64, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 56, in handle_client
    CLIENTS_BY_USERNAME[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 64, in handle_client
    username = message_json.get("userName")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 56, in handle_client
    CLIENTS_BY_USERNAME[message_json["userName"]] = (websocket, timestamp)
KeyError: 'userName'
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 57, in handle_client
    clients_by_username[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd2389136a0>, 1705138990.136034)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 57, in handle_client
    clients_by_username[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd2389136a0>, 1705138990.136034), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fd238913f40>, 1705138990.16347)}
INFO:root:Closed connection with d
INFO:websockets.server:connection closed
INFO:root:Closed connection with dv
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"sdcsdc"}
INFO:root:New user connected: sdcsdc
INFO:root:clients_by_username: {'sdcsdc': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fcf106f36a0>, 1705138998.237291)}
INFO:root:Received message from the client: {"type":"message","userName":"sdcsdc","text":"sdcsdc"}
INFO:root:Received message from the client: {"type":"disconnect","username":"sdcsdc"}
INFO:root:User disconnected: sdcsdc
INFO:root:clients_by_username: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 57, in handle_client
    clients_by_username[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 57, in handle_client
    clients_by_username[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","user_name":"d"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 57, in handle_client
    clients_by_username[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","user_name":"dv"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 57, in handle_client
    clients_by_username[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 57, in handle_client
    clients_by_username[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    user_name = message_json.get("user_name")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 57, in handle_client
    clients_by_user_name[message_json["user_name"]] = (websocket, timestamp)
KeyError: 'user_name'
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 57, in handle_client
    clients_by_username[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7faa90878100>, 1705139236.122592)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect"}
INFO:root:Closed connection with None
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 57, in handle_client
    clients_by_username[message_json["username"]] = (websocket, timestamp)
KeyError: 'username'
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:clients_by_username: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7faa90878100>, 1705139236.122592), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7faa600eab00>, 1705139236.1508968)}
INFO:root:Closed connection with d
INFO:websockets.server:connection closed
INFO:root:Closed connection with dv
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 66, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dfvdfvf"}
INFO:root:New user connected: dfvdfvf
INFO:root:clients_by_username: {'dfvdfvf': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7febb86d36a0>, 1705139242.976924)}
INFO:root:Received message from the client: {"type":"message","userName":"dfvdfvf","text":"dfvdfv"}
INFO:root:Closed connection with None
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 65, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"svsfv"}
INFO:root:New user connected: svsfv
INFO:root:CLIENTS_BY_USERNAME: {'svsfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa228917be0>, 1705139292.6694632)}
INFO:root:Received message from the client: {"type":"message","userName":"svsfv","text":"dfvdfvdfv"}
INFO:root:Received message from the client: {"type":"message","userName":"svsfv","text":"dfvdfv"}
INFO:root:Received message from the client: {"type":"disconnect","username":"svsfv"}
INFO:root:User disconnected: svsfv
INFO:root:CLIENTS_BY_USERNAME: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa2289f0610>, 1705139394.1235058)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa2289f0610>, 1705139394.1235058), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa228917ac0>, 1705139394.136835)}
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 65, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dfvdfvdfv"}
INFO:root:New user connected: dfvdfvdfv
INFO:root:CLIENTS_BY_USERNAME: {'dfvdfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fb928917be0>, 1705139400.973771)}
INFO:root:Received message from the client: {"type":"message","userName":"dfvdfvdfv","text":"dfdvdfvdfv"}
INFO:root:Received message from the client: {"type":"message","userName":"dfvdfvdfv","text":"dffdf"}
INFO:root:Received message from the client: {"type":"disconnect","username":"dfvdfvdfv"}
INFO:root:User disconnected: dfvdfvdfv
INFO:root:CLIENTS_BY_USERNAME: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 65, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"connect","username":"dfvdfv"}
INFO:root:New user connected: dfvdfv
INFO:root:CLIENTS_BY_USERNAME: {'dfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fda809132b0>, 1705139691.037667)}
INFO:root:Received message from the client: {"type":"message","userName":"dfvdfv","text":"dfvdfvdfv"}
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 65, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dfvdfvdfv"}
INFO:root:New user connected: dfvdfvdfv
INFO:root:CLIENTS_BY_USERNAME: {'dfvdfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f99f86afbe0>, 1705139956.875948)}
INFO:root:Received message from the client: {"type":"message","userName":"dfvdfvdfv","text":"dfvfvv"}
INFO:root:Received message from the client: {"type":"disconnect","username":"dfvdfvdfv"}
INFO:root:User disconnected: dfvdfvdfv
INFO:root:CLIENTS_BY_USERNAME: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 65, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dfv"}
INFO:root:New user connected: dfv
INFO:root:CLIENTS_BY_USERNAME: {'dfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fc3589ec2e0>, 1705140035.5085921)}
INFO:root:Received message from the client: {"type":"message","userName":"dfv","text":"dfvfv"}
INFO:root:Received message from the client: {"type":"disconnect","username":"dfv"}
INFO:root:User disconnected: dfv
INFO:root:CLIENTS_BY_USERNAME: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 65, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dfvdfv"}
INFO:root:New user connected: dfvdfv
INFO:root:CLIENTS_BY_USERNAME: {'dfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f7f306d3d30>, 1705140158.010886)}
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 70, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":""}
INFO:root:New user connected: 
INFO:root:CLIENTS_BY_USERNAME: {'': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fb768913d30>, 1705141641.5508952)}
INFO:root:Received message from the client: {"type":"message","userName":"","text":"dfvdfvdfv"}
INFO:root:Received message from the client: {"type":"message","userName":"","text":"dfvdfvdfvdfv"}
INFO:root:Received message from the client: {"type":"disconnect","username":""}
INFO:root:User disconnected: 
INFO:root:CLIENTS_BY_USERNAME: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fb7689ec760>, 1705141732.78685)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fb7689ec760>, 1705141732.78685), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fb768913c10>, 1705141732.799669)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 70, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":""}
INFO:root:New user connected: 
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fb7689ec760>, 1705141732.78685), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fb768913c10>, 1705141732.799669), '': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fb7689eccd0>, 1705141737.4201338)}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 70, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d8913d30>, 1705141787.582329)}
INFO:root:Received message from the client: {"type":"message","userName":"dolev","text":"hey"}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 70, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"shir"}
INFO:root:New user connected: shir
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d8913d30>, 1705141787.582329), 'shir': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d8913280>, 1705141813.145647)}
INFO:root:Received message from the client: {"type":"message","userName":"shir","text":"yyyy"}
INFO:root:Received message from the client: {"type":"message","userName":"dolev","text":"dfnfgn"}
INFO:root:Received message from the client: {"type":"disconnect","username":"dolev"}
INFO:root:User disconnected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'shir': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d8913280>, 1705141813.145647)}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"disconnect","username":"shir"}
INFO:root:User disconnected: shir
INFO:root:CLIENTS_BY_USERNAME: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 70, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ecf40>, 1705142036.232076)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 70, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"s"}
INFO:root:New user connected: s
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ecf40>, 1705142036.232076), 's': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed570>, 1705142038.806098)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 70, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89edab0>, 1705142165.8132648)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed450>, 1705142165.826894)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed450>, 1705142165.826894), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed6c0>, 1705142165.840016)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"s"}
INFO:root:New user connected: s
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed450>, 1705142165.826894), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed6c0>, 1705142165.840016), 's': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed270>, 1705142165.852413)}
INFO:root:Received message from the client: {"type":"disconnect","username":"d"}
INFO:root:User disconnected: d
INFO:root:CLIENTS_BY_USERNAME: {'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed6c0>, 1705142165.840016), 's': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed270>, 1705142165.852413)}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"disconnect","username":"s"}
INFO:root:User disconnected: s
INFO:root:CLIENTS_BY_USERNAME: {'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed6c0>, 1705142165.840016)}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 70, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"connect","username":"dfvdfv"}
INFO:root:New user connected: dfvdfv
INFO:root:CLIENTS_BY_USERNAME: {'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed6c0>, 1705142165.840016), 'dfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed870>, 1705142186.7475982)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 70, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"ddd"}
INFO:root:New user connected: ddd
INFO:root:CLIENTS_BY_USERNAME: {'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed6c0>, 1705142165.840016), 'dfvdfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed870>, 1705142186.7475982), 'ddd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ee020>, 1705142190.5222988)}
INFO:root:Received message from the client: {"type":"disconnect","username":"dfvdfv"}
INFO:root:User disconnected: dfvdfv
INFO:root:CLIENTS_BY_USERNAME: {'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed6c0>, 1705142165.840016), 'ddd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ee020>, 1705142190.5222988)}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"disconnect","username":"ddd"}
INFO:root:User disconnected: ddd
INFO:root:CLIENTS_BY_USERNAME: {'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed6c0>, 1705142165.840016)}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed8d0>, 1705142282.795257)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ed8d0>, 1705142282.795257), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa5d89ede70>, 1705142282.810244)}
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f98109135e0>, 1705143957.2697961)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 81, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 81, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f98109e1270>, 1705143973.188599)}
INFO:root:Received message from the client: {"type":"disconnect","username":"d"}
INFO:root:User disconnected: d
INFO:root:CLIENTS_BY_USERNAME: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f98109e1510>, 1705143983.072098)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:websockets.server:connection closed
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"fvfv"}
INFO:root:New user connected: fvfv
INFO:root:CLIENTS_BY_USERNAME: {'fvfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be86d35e0>, 1705144012.919127)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"fvfv"}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be87a0dc0>, 1705144534.590121)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"fvfv"}
INFO:root:New user connected: fvfv
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be87a0dc0>, 1705144534.590121), 'fvfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be87a0a90>, 1705144534.602709)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be87a0dc0>, 1705144534.590121), 'fvfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be87a0a90>, 1705144534.602709), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be87a10f0>, 1705144534.617381)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:websockets.server:connection closed
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be87a1540>, 1705144663.586886)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"fvfv"}
INFO:root:New user connected: fvfv
INFO:root:CLIENTS_BY_USERNAME: {'fvfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be87a1ea0>, 1705144663.6152651)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:CLIENTS_BY_USERNAME: {'fvfv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be87a1ea0>, 1705144663.6152651), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8be87a21a0>, 1705144663.630243)}
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f883058b4f0>, 1705144703.0556412)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f88306587c0>, 1705144723.58612)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f88306587c0>, 1705144723.58612), 'd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8830659000>, 1705144723.598321)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f88306587c0>, 1705144723.58612), 'd': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8830659000>, 1705144723.598321), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f88306592a0>, 1705144723.6130471)}
INFO:root:Received message from the client: {"type":"disconnect","username":"dolev"}
INFO:root:User disconnected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f8830659000>, 1705144723.598321), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7f88306592a0>, 1705144723.6130471)}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa4186d34f0>, 1705144743.625466)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7ffc489134f0>, 1705145368.88627)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 76, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on [::1]:3001
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 76, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa200913fa0>, 1705145396.177367)}
INFO:root:Received message from the client: {"type":"disconnect","username":"dolev"}
INFO:root:User disconnected: dolev
INFO:root:CLIENTS_BY_USERNAME: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"s"}
INFO:root:New user connected: s
INFO:root:CLIENTS_BY_USERNAME: {'s': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa2009e09d0>, 1705145404.492553)}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 76, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"s"}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"connect","username":"d"}
INFO:root:New user connected: d
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa2009e0f40>, 1705145446.5763588)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"s"}
INFO:root:New user connected: s
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa2009e0f40>, 1705145446.5763588), 's': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa2009e12d0>, 1705145446.591883)}
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dv"}
INFO:root:New user connected: dv
INFO:root:CLIENTS_BY_USERNAME: {'d': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa2009e0f40>, 1705145446.5763588), 's': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa2009e12d0>, 1705145446.591883), 'dv': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fa2009e1570>, 1705145446.60636)}
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
INFO:websockets.server:server listening on 127.0.0.1:3001
INFO:websockets.server:server listening on [::1]:3001
INFO:root:WebSocket server listening on ws://localhost:3001
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 71, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"dolev"}
INFO:root:New user connected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fbf80913d30>, 1705145499.066483)}
INFO:root:Received message from the client: {"type":"message","userName":"dolev","text":"dfvdfvdfv"}
INFO:websockets.server:connection open
INFO:root:Client disconnected.
ERROR:websockets.server:connection handler failed
Traceback (most recent call last):
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 236, in handler
    await self.ws_handler(self)
  File "/Users/dolevgabay/anaconda3/lib/python3.10/site-packages/websockets/legacy/server.py", line 1175, in _ws_handler
    return await cast(
  File "/Users/dolevgabay/Desktop/raincheck/chat/server.py", line 71, in handle_client
    username = message_json.get("username")
UnboundLocalError: local variable 'message_json' referenced before assignment
INFO:websockets.server:connection closed
INFO:websockets.server:connection open
INFO:root:Received message from the client: {"type":"connect","username":"shir"}
INFO:root:New user connected: shir
INFO:root:CLIENTS_BY_USERNAME: {'dolev': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fbf80913d30>, 1705145499.066483), 'shir': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fbf80912650>, 1705145512.558699)}
INFO:root:Received message from the client: {"type":"message","userName":"shir","text":"gbfgbfgb"}
INFO:root:Received message from the client: {"type":"message","userName":"shir","text":"fff"}
INFO:root:Received message from the client: {"type":"message","userName":"dolev","text":"jmjm"}
INFO:root:Received message from the client: {"type":"disconnect","username":"dolev"}
INFO:root:User disconnected: dolev
INFO:root:CLIENTS_BY_USERNAME: {'shir': (<websockets.legacy.server.WebSocketServerProtocol object at 0x7fbf80912650>, 1705145512.558699)}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:Received message from the client: {"type":"disconnect","username":"shir"}
INFO:root:User disconnected: shir
INFO:root:CLIENTS_BY_USERNAME: {}
INFO:root:Client disconnected.
INFO:websockets.server:connection closed
INFO:root:WebSocket server interrupted. Closing connections...
INFO:root:WebSocket server thread joined. Exiting.
